site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,175 @@
+
+{% aside 'Additional injection options' %}
+`Injectors` are capable of injecting more than just properties into a widget. A `getChildren` mapping function can also be provided to the `Container`, allowing children to be injected into the widget as well. Just like the `getProperties` function, the `getChildren` function will receive the children from the parent widget and the injector. It then returns the final array of children that will be provided to the contained widget.
+{% endaside %}
+
+The `getProperties` function receives two parameters. The first is the `payload` of the `injector` instance, for the demo application this is the data passed when creating the `ApplicationContext`. The second is the `properties` that have been passed to the container via the normal mechanism, `w(Container, properties)`, the properties will implement the properties interface defined by the wrapped widget (for example `WorkerContainerProperties`). The `getProperties` function must then return an object that holds the properties that will be passed to the widget itself. In this example, we are ignoring the properties provided by the parent and returning the `workerData` stored by the `ApplicationContext`. More advanced use cases where both sources are used to generate the properties are also possible.
+
+{% instruction 'Finish the `WorkerContainerContainer` by adding the following code.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerContainerContainer.ts' lines:10-12 %}
+
+These final lines define the actual `WorkerContainerContainer` class and export it. The `Container` function creates the class by accepting three parameters - the widget's class definition (alternatively, a widget's registry key can be used), the registry key for the `Injector`, and an object literal that provides the mapping functions used to reconcile the two sets of properties and children that the container can receive (one from the `Injector` and one from the parent widget). The returned class is, in fact, a widget (i.e. it descends from `WidgetBase`) and, therefore may be used just like any other widget.
+
+The other container that we need is the `WorkerFormContainer`.
+
+{% instruction 'Add the following code to the `WorkerFormContainer` module in the `containers` sub-package.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerFormContainer.ts' %}
+
+This module is almost identical to the `WorkerContainerContainer` except that additional properties are required by the `WorkerForm` to allow it to respond to user interactions with the form. The `ApplicationContext` contains two methods for managing these events - `onFormInput` and `onFormSave`. These methods need to be passed into the `WorkerForm` to handle the events, but they need to execute in the context of the `ApplicationContext`. To handle this, `bind` is called on each of the methods to explicitly set their execution contexts.
```

Comment:
"except that additional properties are required by the" -> "except for additional properties that are required by the"